### PR TITLE
Feature/hide graph column

### PR DESCRIPTION
**Pull request by ericf-axosoft**
# Summary of changes (Gitlens side)
- Added option to hide/show columns in the status bar.
- We can only hide sha/author/date columns like in GKC. 

https://user-images.githubusercontent.com/90025366/190666432-676a4e2e-98d5-4947-b7ba-aeac95d1b87b.mov

# Notes
This PR works with: https://github.com/gitkraken/GitKrakenComponents/pull/64

# Task
https://github.com/gitkraken/GitKrakenComponents/issues/58

### CODE REVIEW

File: src/webviews/apps/plus/graph/GraphWrapper.tsx [hunk 1]
```diff
@@ -58,11 +58,18 @@ const getStyleProps = (
...
+	commitMessageZone: { width: 130, isHidden: false },
+	commitZone: { width: 170, isHidden: false },
+	refZone: { width: 150, isHidden: false },
+};
+
+const getGraphColConfigModel = (columnSettings: GKGraphColumnSetting): GraphColumnConfig => {
+	return {
+		width: columnSettings.width,
+		isHidden: columnSettings.isHidden,
+	};
```
Comment (d13) [inline src/webviews/apps/plus/graph/GraphWrapper.tsx:72]: Does `GKGraphColumnSetting` have more properties than `width` and `isHidden`? If not, this function is likely not needed.

File: src/webviews/apps/plus/graph/GraphWrapper.tsx [hunk 2]
```diff
@@ -517,6 +567,50 @@ export function GraphWrapper({
 							showing {graphRows.length} item{graphRows.length ? 's' : ''}
 						</span>
 					)}
+					<div className="actioncombo">
+						<button
+							type="button"
+							aria-controls="repo-columnsettings-list"
+							aria-expanded={columnSettingsExpanded}
+							aria-haspopup="listbox"
+							id="repo-actioncombo-label"
```
Comment (d13) [inline src/webviews/apps/plus/graph/GraphWrapper.tsx:576]: using same `repo-` prefix for ids as the repo combobox

File: src/webviews/apps/plus/graph/GraphWrapper.tsx [hunk 3]
```diff
@@ -517,6 +567,50 @@ export function GraphWrapper({
...
+							className="actioncombo__label"
+							role="combobox"
+							aria-activedescendant={
+								columnSettingsExpanded
+									? `repo-actioncombo-item-${reposList.findIndex(
+											item => item.path === currentRepository?.path,
+									  )}`
+									: undefined
```
Comment (d13) [inline src/webviews/apps/plus/graph/GraphWrapper.tsx:584]: this should be looking against the graph columns

File: src/webviews/apps/plus/graph/GraphWrapper.tsx [hunk 4]
```diff
@@ -517,6 +567,50 @@ export function GraphWrapper({
...
+							aria-labelledby="repo-actioncombo-label"
+						>
+							{
+								Object.entries(graphColumns).map(([graphZoneType, column]) => column.hideable && (
+									<span
+										className="actioncombo__item"
+										role="option"
+										data-value={graphZoneType}
+										id={`repo-actioncombo-item-${graphZoneType}`}
+										key={`repo-actioncombo-item-${graphZoneType}`}
```
Comment (d13) [inline src/webviews/apps/plus/graph/GraphWrapper.tsx:604]: same as above for the `repo-` prefix

**Review by d13 — CHANGES_REQUESTED** [review]

File: src/webviews/apps/plus/graph/GraphWrapper.tsx (hunk 1 above)
Comment (ericf-axosoft) [reply src/webviews/apps/plus/graph/GraphWrapper.tsx]: Done in commit: `Refactoring: removed "getGraphColConfigModel" function`

File: src/webviews/apps/plus/graph/GraphWrapper.tsx (hunk 2 above)
Comment (ericf-axosoft) [reply src/webviews/apps/plus/graph/GraphWrapper.tsx]: Done in commit: `Fixed ids/keys prefix used for the columns combo`

File: src/webviews/apps/plus/graph/GraphWrapper.tsx (hunk 3 above)
Comment (ericf-axosoft) [reply src/webviews/apps/plus/graph/GraphWrapper.tsx]: Done in commit: `Fixed ids/keys prefix used for the columns combo`

File: src/webviews/apps/plus/graph/GraphWrapper.tsx (hunk 4 above)
Comment (ericf-axosoft) [reply src/webviews/apps/plus/graph/GraphWrapper.tsx]: Done in commit: `Fixed ids/keys prefix used for the columns combo`

**Review by axosoft-ramint — APPROVED** [review]
Not as big of an expert on this side of things but it seemed to work ok in my tests (tested toggling visibility on columns and closing/reopening the graph to make sure it remembered my visibility state).

**Pull request merged**